sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-processing-steps/validate-cslp-lifetime-maximum-check-step.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from "@nestjs/common";
+import { ECertProcessStep } from "./e-cert-steps-models";
+import { ProcessSummary } from "@sims/utilities/logger";
+import { EntityManager } from "typeorm";
+import { EligibleECertDisbursement } from "../disbursement-schedule.models";
+import { ECertGenerationService } from "../e-cert-generation.service";
+import { CANADA_STUDENT_LOAN_PART_TIME_AWARD_CODE } from "@sims/services/constants";
+import { StudentLoanBalanceSharedService } from "@sims/services/student-loan-balance/student-loan-balance-shared.service";
+import { StudentLoanBalance } from "@sims/sims-db";
+
+/**
+ * Fetch the latest  balance for the student and validate if CSLP has reached lifetime maximum.
+ */
+@Injectable()
+export class ValidateCSLPLifetimeMaximumCheckStep implements ECertProcessStep {
+  constructor(
+    private readonly eCertGenerationService: ECertGenerationService,
+    private readonly studentLoanBalanceSharedService: StudentLoanBalanceSharedService,
+  ) {}
+
+  /**
+   * Validate if CSLP disbursed exceeded the lifetime maximums.
+   * @param eCertDisbursement eligible disbursement to be potentially added to an e-Cert.
+   * @param entityManager used to execute the commands in the same transaction.
+   * @param log cumulative log summary.
+   */
+  async executeStep(
+    eCertDisbursement: EligibleECertDisbursement,
+    entityManager: EntityManager,
+    log: ProcessSummary,
+  ): Promise<boolean> {
+    log.info("Validate CSLP Lifetime Maximum.");
+    //Fetch lifetime maximum of CSLP from the workflow data.
+    const lifetimeMaximumsCSLP =
+      await this.eCertGenerationService.getCSLPLifetimeMaximums(
```

Comment:
This can potentially be moved to the main query that returns the values for `EligibleECertDisbursement`.
Since it is a data that will be needed for every single validation and it will never change between the calculation of the different records maybe it would be worth to be retrieved from the main query.
Thoughts?